src/mavlink-router/endpoint.cpp

```diff
@@ -332,16 +335,16 @@ bool Endpoint::accept_msg(int target_sysid, int target_compid, uint8_t src_sysid
     if (!allowed_by_filter(msg_id))
         return false;
 
-    // Message is broadcast on sysid: accept msg
+    // Message is broadcast on sysid or sysid is non-existent: accept msg
     if (target_sysid == 0 || target_sysid == -1)
         return true;
 
     // This endpoint has the target of message (sys and comp id): accept
     if (target_compid > 0 && has_sys_comp_id(target_sysid, target_compid))
         return true;
 
-    // This endpoint has the target of message (sysid, but compid is broadcast): accept
-    if (has_sys_id(target_sysid))
+    // This endpoint has the target of message (sysid, but compid is broadcast or non-existent): accept
+    if ((target_compid == 0 || target_compid == -1) && has_sys_id(target_sysid))
```

Comment:
@julianoes: if you are curious, this is the fix for those unwanted messages we observed. I realized it was fixed upstream only after I identified it :confused: